Packs/PrismaSaasSecurity/Integrations/SaasSecurityEventCollector/SaasSecurityEventCollector.py

```diff
@@ -0,0 +1,254 @@
+        headers = {
+            'accept': 'application/json',
+            'Content-Type': 'application/x-www-form-urlencoded; charset=ISO-8859-1',
+            'Authorization': f'Basic {base64_encoded_creds}',
+        }
+        params = {
+            'grant_type': 'client_credentials',
+            'scope': 'api_access',
+        }
+        token_response = self._http_request('POST', url_suffix='/oauth/token', params=params, headers=headers)
+        return token_response.get('access_token'), token_response.get('expires_in')
+
+    def get_events_request(self):
+        """
+        Get up to 100 event logs.
+        """
+        return self.http_request(
+            'GET',
+            url_suffix='/api/v1/log_events_bulk',
+            resp_type='response',
+            ok_codes=[200, 204],
+        )
+
+    def get_event_request(self):
+        """
+        Get only a single event log - used only for the test-module to verify log-access api scope.
+        """
+        return self.http_request(
+            'GET',
+            url_suffix='/api/v1/log_events',
+            resp_type='response',
+            ok_codes=[200, 204],
+        )
+
+
+def is_token_expired(token_initiate_time: float, token_expiration_seconds: float) -> bool:
+    """
+    Check whether a token has expired. a token considered expired if it has been reached to its expiration date in
+    seconds minus a minute.
+
+    for example ---> time.time() = 300, token_initiate_time = 240, token_expiration_seconds = 120
+
+    300.0001 - 240 < 120 - 60
+
+    Args:
+        token_initiate_time (float): the time in which the token was initiated in seconds.
+        token_expiration_seconds (float): the time in which the token should be expired in seconds.
+
+    Returns:
+        bool: True if token has expired, False if not.
+    """
+    return time.time() - token_initiate_time >= token_expiration_seconds - 60
+
+
+def validate_limit(limit: int):
+    """
+    Validate that the limit/max fetch is a number divisible by the MAX_EVENTS_PER_REQUEST (100).
+    """
+    if limit % MAX_EVENTS_PER_REQUEST != 0:
+        raise DemistoException(f'fetch limit parameter should be divisible by {MAX_EVENTS_PER_REQUEST}')
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client):
+    """
+    Testing we have a valid connection to Saas-Security.
+
+    The reason not testing fetching events is that if we fetch it from the api we will lose those requests.
+    """
+    response = client.get_event_request()
+    if response.status_code in (200, 204):
+        return 'ok'
+
+
+def saas_security_get_events_command(client: Client, args: Dict) -> Union[str, CommandResults]:
+    """
+    Fetches events from the saas-security queue and return them to the war-room.
+    in case should_push_events is set to True, they will be also sent to XSIAM.
+    """
+    limit = arg_to_number(args.get('limit')) or 100
```

Comment:
lets add this logic in the main

Reply:
may i ask why? usually we do the logic of parsing arguments in the command function

Reply:
since this is an override of the param it should be obvious, and done in line.Its also cleaner not to need to call validate twice

Reply:
done